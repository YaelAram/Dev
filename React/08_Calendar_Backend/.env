PORT=8080
PRIVATE_KEY=BlueCoffeeBlackie
DB_CNN=mongodb+srv://NodeUser:hpaHnnprt3v06bqI@nodetest.hxx05.mongodb.net/mern_calendar
