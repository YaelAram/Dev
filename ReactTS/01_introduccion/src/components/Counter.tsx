import { useState } from "react"

export function Counter() {
  const [counter, setCounter] = useState(0);

  const handleAdd = (step: number = 1): void => {
    setCounter(counter + step);
  };

  return (
    <>
      <span>Valor: {counter}</span>
      <br />
      <button type="button" className="btn btn-outline-primary mt-2" onClick={() => handleAdd()}>+1</button>
      <button type="button" className="btn btn-outline-primary ms-3 mt-2" onClick={() => handleAdd(2)}>+2</button>
      <button type="button" className="btn btn-outline-danger ms-3 mt-2" onClick={() => setCounter(0)}>Reset</button>
    </>
  )
}
