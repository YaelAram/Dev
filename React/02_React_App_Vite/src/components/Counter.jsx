import React, { useReducer } from 'react';
import PropTypes from 'prop-types';

const handdleCounter = ( currentValue, { type, init } ) => {
    switch( type ){
        case 'add': return currentValue + 1;
        case  'substract': return currentValue - 1;
        default: return init;
    }
};

function Counter( { value } ){
    const [ counter, dispath ] = useReducer( handdleCounter, value );

    return (
        <React.Fragment>
            <h1>Counter App</h1>
            <h2>{ counter }</h2>
            <input type="button" aria-label="btn-substract" value="Substract 1" onClick={ () => dispath( { type: 'substract' } ) }/>
            <input type="button" aria-label="btn-reset" value="Reset" onClick={ () => dispath( { type: 'reset', init: value } ) }/>
            <input type="button" aria-label="btn-add" value="Add 1" onClick={ () => dispath( { type: 'add' } ) }/>
        </React.Fragment>
    );
};

Counter.propTypes = {
    value: PropTypes.number.isRequired
};

export default Counter;
