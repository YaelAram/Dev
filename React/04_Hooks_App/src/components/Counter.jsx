import React, { useState } from 'react';

export function Counter() {
    const [ state, setState ] = useState( {
        counter1: 10,
        counter2: 20,
        counter3: 30
    } );

    const { counter1, counter2, counter3 } = state;

    const handleOnClick = () => {
        setState( () => {
            const stateCopy = { ...state };
            for ( const key in stateCopy ) stateCopy[ key ] += 1;
            return stateCopy;
        } );
    };

    return (
        <React.Fragment>
            <h3>Counter 1: { counter1 }</h3>
            <h3>Counter 2: { counter2 }</h3>
            <h3>Counter 3: { counter3 }</h3>
            <input type="button" value="+1" onClick={ handleOnClick } />
        </React.Fragment>
    );
};
