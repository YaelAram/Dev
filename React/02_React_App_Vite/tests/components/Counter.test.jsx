import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react';

import Counter from '../../src/components/Counter';

describe( 'Counter Component', () => {
    const value = 0;

    test( 'Counter snapshot', () => {
        const { container } =  render( <Counter value={ value }/> );
        expect( container ).toMatchSnapshot();
    } );

    test( 'Check value is on H2', () => {
        render( <Counter value={ value }/> );
        expect( screen.getByRole( 'heading', { level: 2 } ).innerHTML ).toContain( String( value ) );
    } );

    test( 'Increment button', () => {
        render( <Counter value={ value }/> );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-add' } ) );
        expect( screen.getByRole( 'heading', { level: 2 } ).innerHTML ).toContain( '1' );
    } );

    test( 'Decrement button', () => {
        render( <Counter value={ value }/> );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-substract' } ) );
        expect( screen.getByRole( 'heading', { level: 2 } ).innerHTML ).toContain( '-1' );
    } );

    test( 'Reset button', () => {
        render( <Counter value={ value }/> );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-substract' } ) );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-substract' } ) );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-substract' } ) );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-add' } ) );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-reset' } ) );
        expect( screen.getByRole( 'heading', { level: 2 } ).innerHTML ).toContain( String( value ) );
    } );
} );
