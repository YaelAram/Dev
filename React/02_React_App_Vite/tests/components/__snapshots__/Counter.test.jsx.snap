// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Counter snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    0
  </h2>
  <input
    aria-label="btn-substract"
    type="button"
    value="Substract 1"
  />
  <input
    aria-label="btn-reset"
    type="button"
    value="Reset"
  />
  <input
    aria-label="btn-add"
    type="button"
    value="Add 1"
  />
</div>
`;
